oracles/src/repositories/web3/base.py

```diff
@@ -21,6 +21,22 @@ def __init__(self) -> None:
             "errors": 0,
         }
 
+    async def _find_first_unprocessed(self, count, is_processed_func):
+        low = 0
+        high = count
+        while low < high:
+            index = (low + high) // 2
+            try:
+                is_prompt_processed = await is_processed_func(index)
+                if is_prompt_processed:
+                    low = index + 1
+                else:
+                    high = index
+            except Exception as e:
+                print(f"Skipping unreadable job at index {index}: {e}")
+                low = index + 1
```

Comment:
hmm can this potentially cause issues? If the first unprocessed func is lower than index but the current index has some issues?